R/html_dependencies.R

```diff
@@ -88,16 +92,35 @@ bootstrap_dependencies <- function(theme) {
 resolve_theme <- function(theme) {
   # theme = NULL means no Bootstrap
   if (is.null(theme)) return(theme)
+
   # Bootstrap/Bootswatch 3 names (backwards-compatibility)
   if (is.character(theme)) {
     if (length(theme) != 1) {
       stop2("`theme` must be character vector of length 1.")
     }
-    if (theme %in% c("bootstrap", "default")) {
-      return("bootstrap")
+    if (identical(theme, "default")) {
+      theme <- "bootstrap"
+    }
+
+    # special handling triggered when bootstrap folder removed from installation folder
+    # In this case, bslib is used as default.
+    bslib_forced_mode <- getOption("rmarkdown.bslib_forced_mode", FALSE)
+    if (bslib_forced_mode || !dir.exists(pkg_file("rmd/h/bootstrap"))) {
+      # we are in special bslib default mode, theme is tweaked to be a list
+      if (!is_available("bslib")) {
```

Comment:
In both cases, I think we can assume that users know what they are doing, so I wonder if this explicit error message is necessary, i.e., if they set the option to `TRUE` or delete the bootstrap folder, they must already know that they will need to install **bslib**.

Another factor to consider is that **bslib** will be a hard dependency in the future. By that time, this check won't be necessary.

Reply:
I'll leave as is for now. and remove later when bslib will be hard dependency and dependening on feedback we have. 

Thanks!